### ISSUE TITLE
Move to opentelemetry-dotnet-contrib?

### ISSUE THREAD

**Issue by glucaci** [jbogard/MongoDB.Driver.Core.Extensions.DiagnosticSources#10]
Hi,

Would you like to move this package to the [opentelemetry-dotnet-contrib](https://github.com/open-telemetry/opentelemetry-dotnet-contrib) as instrumentation package?

Thanks!

**Comment by jbogard** [comment]
Honestly, I'd rather this be in the MongoDB library directly.

**Comment by simaoribeiro** [comment]
Hi there @jbogard . Do you know if there are any updates regarding the inclusion of these OTEL capabilities in Mongo C# driver?

**Comment by jbogard** [comment]
I opened a JIRA issue a while back: https://jira.mongodb.org/browse/CSHARP-3124

Feel free to upvote/comment there!

**Comment by simaoribeiro** [comment]
Thanks. Seems like it will take a while to have official support from MongoDB team

**Comment by glucaci** [comment]
So we don't want to move to contrib repo from opentelemenetry?

With integration in MongoDB library from my experience can take years.

Thanks!

**Comment by jbogard** [comment]
No, I don't think so, unless someone else wants to take over the maintenance of it over on that repo. I don't really have the time to get involved with the amount of changes, the quality bar, and release schedule they have.

**Comment by jbogard** [comment]
OK so it's been 3 years now with zero movement in the C# driver. Tagging @lucecarter from the Mongo side and @martinjt from the Otel contrib side.

I don't care where it goes but at this point certainly not here.

**Comment by LuceCarter** [comment]
From a MongoDB perspective, its not ruled out that we will take it over, it will just take time.
Our teams are currently doing a whole OTEL initiative to have client side OTEL support across all drivers which is currently in technical design phase so its standard and consistent.
Because it is across all languages, it is a longer process amongst other priorities. I don't have a timeline as DevRel isn't involved but I have raised to the C# team that this library is looking for a home so as the technical design is completed, I can probably push for a discussion about how we can incorporate this and if it is something we can take over. I just didn't want you sat around wondering our current position 🙂
Happy to keep this thread up to date as I know more.

**Comment by martinjt** [comment]
We can move this functionality into contrib, which is a better home, but I'm reluctant to do that without a planned timeline from MongoDB.

Last I saw, it had been moved to next year in the Jira tracker at Mongo, which would make sense to move it to contrib (although I'm disappointed with that timeline for such an important feature).

**Comment by damieng** [comment]
Hi!

The MongoDB .NET/C# team is actively looking at this. Based on our initial analysis and our Cross Drivers specifications, the implementation will involve a little more than what this package entails. We're aiming to add this to the .NET/C# driver before the end of July this year.

PS: FY26 implies this calendar year for our timeline.

**Comment by jbogard** [comment]
Ah good news! Looking forward to archiving this repo and deprecating the package.
